### ISSUE TITLE
Access ShareClient using ClientSecretCredential

### ISSUE THREAD

**Issue by TadasBaltrusaitis** [Azure/azure-sdk-for-python#11492]
Currently it is possible to access `azure.storage.blob.ContainerClient` using `azure.identity.ClientSecretCredential`. However, unless I am missing something, to access `azure.storage.fileshare.ShareClient` we need to provide an SAS token or an account key (which might be difficult to acquire).

It would be great if we could pass the same credentials that we use to access `ContainerClient` to access the `ShareClient` (it would also increase the consistency between the interfaces).

**Comment by xiafu-msft** [comment]
Hi @TadasBaltrusaitis 

Thanks for reaching out. We will have a discussion with service side and see if they have plan to add AAD auth support!

**Comment by jalauzon-msft** [comment]
Hi @TadasBaltrusaitis Tadas, I realize you opened this quite some time ago and I apologize it's been so long without an update, but we have been waiting on the service to add support for this. I'm happy to report they are working on adding this support and we should be adding it to an upcoming release of the SDK.

As a housekeeping item, I'm going to close this issue as a duplicate of #13033 as it is asking for the same thing, Files support for AAD auth (as that one has a few more people following it). I will comment an update there and use that one to track the upcoming work for Files OAuth. Thanks.
